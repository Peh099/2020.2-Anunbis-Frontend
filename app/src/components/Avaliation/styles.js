import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: ${(props) => (props.direction ? props.direction : 'column')};
  width: ${(props) => (props.width ? props.width : '372')};
  height: ${(props) => (props.heigth ? props.heigth : 'fit-content')};
  align-self: flex-start;
  justify-content: center;
  border-bottom-right-radius: 2%;
  border-bottom-left-radius: 2%;
  background-color: ${(props) =>
    props.backColor ? props.backColor : 'var(--lightWhite)'};

  Form {
    align-items: ${(props) => (props.align ? props.align : '')};
    align-items: flex-start;
    p,
    Input,
    select,
    button {
      margin-top: 10px;
      margin-bottom: 10px;
    }

    button {
      display: flex;
      justify-content: center;
      align-items: center;
      border: var(--black);
      @media (max-width: 460px) {
        font-size: 11px;
        width: 20vw;
        overflow: hidden;
      }
    }

    textarea {
      padding: 10px;
    }
  }
`;
export const Grades = styled.div`
  display: grid;
  grid-template-columns: 170px 150px;
  font-size: 13px;
  margin-bottom: 10px;

  label {
    margin-top: 5px;
  }
`;

export const NameProfessor = styled.p`
  color: rgba(61, 58, 58, 0.603);
  border-bottom: 1px solid #000000;
  width: 252px;
  text-align: left;
  font-size: 15px;
`;

export const TxtArea = styled.textarea`
  width: 330px;
  height: 70px;
  border-radius: 10px;
  background-color: var(--lightWhite);
  resize: none;

  &:focus {
    outline-width: 0;
  }

  @media (max-width: 460px) {
    width: 75vw;
  }
`;
