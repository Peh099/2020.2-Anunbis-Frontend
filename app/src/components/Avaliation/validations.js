import * as yup from 'yup';

const schema = yup.object().shape({
  id_course: yup.string().required('Nome da disciplina deve ser preenchido.'),

  didactic: yup.array().min(1, 'Dê ao menos uma estrela.'),

  metod: yup.array().min(1, 'Dê ao menos uma estrela.'),

  avaliations: yup.array().min(1, 'Dê ao menos uma estrela.'),

  disponibility: yup.array().min(1, 'Dê ao menos uma estrela.'),

  comments: yup
    .string()
    .trim()
    .required('O comentário deve ser preenchido.')
    .min(10, 'O comentário deve conter pelo menos dez caracteres.')
    .max(254, 'O comentário deve ter no máximo 254 caracteres.'),
});

export default schema;
