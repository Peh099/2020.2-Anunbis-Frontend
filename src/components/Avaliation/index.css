.avaliation {
    width: 20%;
    height: 70%;
    text-align: center;
    initial-letter-align: initial;
    background-color: #FFFDE7;
    border-bottom-right-radius: 2%;
    border-bottom-left-radius: 2%;
}

.avaliation .title {
    width: 100%;
    height: 5%;
    background-color: #212121;
    color: #FFFDE7;
    display: flex;
    align-items: center;
    justify-content: center;
}

.avaliation .title .buttonClose{
    border-radius: 0%;
    height: 100%;
    background-color: red;
}
.avaliation .title .buttonClose:hover{
    background: rgba(255, 0, 0, 0.493);
}

.avaliation .avaliationContent .typePost .button{
    background-color: #FFF9C4;
}
.avaliation .avaliationContent  .typePost .button:hover{
    background: #969681;
}
.avaliation .avaliationContent .typePost .button.selected{
    background: #969681;
}

.avaliation .avaliationContent .buttonPost{
    background-color: #26A69A;
}
.avaliation .avaliationContent .buttonPost:hover{
    background: #26a69965;
}

.avaliation .avaliationContent .typePost {
    padding-top: 10%;
    text-align: center;
}

.avaliation .avaliationContent .commentsPost{
    padding-top: 10%;
    padding-bottom: 50% ;
    padding-left: 10%;
    text-align: center;
    width: 80%;  
}
.avaliation .avaliationContent .commentsPost textarea{
    padding-top: 5%;
    resize: none;
    width: 80%;
    height: 100%;
    border-radius: 10%;
}


/* 


.avaliation Input{
    width: 80%;
    background-color: #FFFDE7;
    padding-bottom: 1%;
}
 */
