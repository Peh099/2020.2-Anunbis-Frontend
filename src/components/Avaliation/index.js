import React from 'react';
import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import styled from 'styled-components';

import schema from './validations';
import Input from '../Input';
import Select from '../Select';
import Button from '../Button';
import Form from '../Form';
import FeedPopup from '../FeedPopup';
import api from '../../services/Api';

const Container = styled.div`
    display: flex;
    flex-direction: ${props => props.direction ? props.direction : "column"};

    width: ${props => props.width ? props.width : "372"};
    height: ${props => props.heigth ? props.heigth : "360px"};

    align-self: flex-end;
    justify-content: center;
    
    border-bottom-right-radius: 2%;
    border-bottom-left-radius: 2%;
    background-color: ${props => props.backColor ? props.backColor : "#FFFDE7"};

    Form{
        align-items: ${props => props.align ? props.align : ""};
        align-items: flex-start;
        p, Input{
            margin-top: 10px;   
        }
    }
`
const NameProfessor = styled.p`
    color: rgba(61, 58, 58, 0.603);
    border-bottom: 1px solid #000000;
    width: 252px;
    text-align: left;
    font-size: 15px;
`

const Title = styled.div`
    width: 50px;
    display:flex;
    color: #FFFDE7;
    margin-left: auto;
    margin-right: auto;
`

const TxtArea = styled.textarea`
    width: 330px;
    height: 70px;
    border-radius: 10px;
    background-color: #FFFDE7;
    resize: none;

    &:focus{
        outline-width: 0;
    }
`
export default function Avaliation({
    close,
    reg_student,
    id_professor,
    name_professor,
    disciplines
    }) {
        function disciplines_Options (disciplines) {
            const disciplinesArray = []
            disciplines?.map((dis) => disciplinesArray.push({id:dis.discipline_code,name:dis.name}))
            return(
                disciplinesArray
            );       
        } 
    const { register, handleSubmit, errors } = useForm({
        resolver: yupResolver(schema),
    });
    const [isAnonymous,setIsAnonymous] = React.useState(false);
    
   function onSubmit(data) {
        const body = {
            reg_student: parseInt(reg_student),
            id_professor: parseInt(id_professor),
            content: data.comments,
            rating: parseFloat(data.note),
            discipline_code: data.id_course,
            is_anonymous: isAnonymous
         }

         api.post("/post", body)
         .then(response => {
             if(response.status === 201){
                close();
             }
         })
        }
        
  return (
    <FeedPopup close={close}>
          <Container>
              <Form onSubmit={handleSubmit(onSubmit)}>
                  <NameProfessor>{name_professor}</NameProfessor>
                    <Form.Field errorMsg={errors.id_course?.message}><Select id="diciplines" backColor="#FFFDE7" text="Selecione um Curso" options={disciplines_Options(disciplines)} name="id_course" register={register} /></Form.Field> 
                    <Form.Field errorMsg={errors.note?.message}><Input type="number" step="0.1" text="Nota" name="note" register={register} width="90px"/></Form.Field> 
                  <p>Postagem:</p>
                  <Container direction="row" heigth="30px" align="center">
                      <Button type="button" padding="7px 10px" text="PÚBLICA" backColor={isAnonymous ? "#FFF9C4" : "#969681"} onClick={() => setIsAnonymous(false)} />
                      <Button type="button" padding="7px 10px" text="ANÔNIMA" backColor={isAnonymous ? "#969681" : "#FFF9C4"} onClick={() => setIsAnonymous(true)} />
                  </Container>
                  <p>Descrição/Comentários:</p>
                    <Form.Field errorMsg={errors.comments?.message}><TxtArea name="comments" ref={register} /></Form.Field> 
                <Form.Footer><Button type="submit" text="POSTAR" backColor="#26A69A" /></Form.Footer>
              </Form>
          </Container>
      </FeedPopup>
  );
}
