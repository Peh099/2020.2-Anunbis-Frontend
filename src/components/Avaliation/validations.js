import * as yup from 'yup';

const schema = yup.object().shape({
    nameProfessor: yup
    .string()
    .trim()
    .min(2, "Nome do professor deve conter pelo menos dois caracteres.")
    .max(254, "Nome do professor deve ter no máximo 254 caracteres.")
    .required("Nome do professor deve ser preenchido."),

    nameDiscipline: yup
    .string()
    .trim()
    .min(2, "Nome da disciplina deve conter pelo menos dois caracteres.")
    .max(254, "Nome da disciplina deve ter no máximo 254 caracteres.")
    .required("Nome da disciplina deve ser preenchido."),

    note: yup
    .string()
    .trim()
    .max(2, "Nota deve ser de 0 a 10")
    .matches(/^[0-9]*$/, "Nota deve conter somente números.")
    .required("Nota deve ser preenchida."),
    
    comments: yup
    .string()
    .trim()
    .min(10, "O comentário deve conter pelo menos dois caracteres.")
    .max(254, "O comentário deve ter no máximo 254 caracteres.")
    .required("O comentário deve ser preenchido."),

})

export default schema;