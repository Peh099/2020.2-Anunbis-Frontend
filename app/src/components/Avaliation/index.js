import React from 'react';
import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import { Container, NameProfessor, TxtArea } from './styles'

import schema from './validations';
import Input from '../Input';
import Select from '../Select';
import Button from '../Button';
import Form from '../Form';
import FeedPopup from '../FeedPopup';
import api from '../../services/Api';

export default function Avaliation({
    close,
    reg_student,
    id_professor,
    name_professor,
    disciplines
    }) {
        function disciplines_Options (disciplines) {
            const disciplinesArray = []
            disciplines?.map((dis) => disciplinesArray.push({id:dis.discipline_code,name:dis.name}))
            return(
                disciplinesArray
            );       
        } 
    const { register, handleSubmit, errors } = useForm({
        resolver: yupResolver(schema),
    });
    const [isAnonymous,setIsAnonymous] = React.useState(false);
    
   function onSubmit(data) {
        const body = {
            reg_student: parseInt(reg_student),
            id_professor: parseInt(id_professor),
            content: data.comments,
            rating: parseFloat(data.note),
            discipline_code: data.id_course,
            is_anonymous: isAnonymous
         }

         api.post("/post", body)
         .then(response => {
             if(response.status === 201){
                close();
             }
         })
        }
        
  return (
    <FeedPopup title="Avaliação"close={close}>
          <Container>
              <Form onSubmit={handleSubmit(onSubmit)}>
                  <NameProfessor>{name_professor}</NameProfessor>
                    <Form.Field errorMsg={errors.id_course?.message}><Select id="diciplines" backColor="#FFFDE7" text="Selecione um Curso" options={disciplines_Options(disciplines)} name="id_course" register={register} /></Form.Field> 
                    <Form.Field errorMsg={errors.note?.message}><Input type="number" step="0.1" text="Nota" name="note" register={register} width="90px"/></Form.Field> 
                  <p>Postagem:</p>
                  <Container direction="row" heigth="30px" align="center">
                      <Button type="button" padding="7px 10px" text="PÚBLICA" backColor={isAnonymous ? "#FFF9C4" : "#969681"} onClick={() => setIsAnonymous(false)} />
                      <Button type="button" padding="7px 10px" text="ANÔNIMA" backColor={isAnonymous ? "#969681" : "#FFF9C4"} onClick={() => setIsAnonymous(true)} />
                  </Container>
                  <p>Descrição/Comentários:</p>
                    <Form.Field errorMsg={errors.comments?.message}><TxtArea name="comments" ref={register} /></Form.Field> 
                <Form.Footer><Button type="submit" text="POSTAR" backColor="#26A69A" /></Form.Footer>
              </Form>
          </Container>
      </FeedPopup>
  );
}
