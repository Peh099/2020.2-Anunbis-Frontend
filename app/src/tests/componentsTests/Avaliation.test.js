import renderer from 'react-test-renderer';
import Avaliation from '../../components/Avaliation';
import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe('Snapshot Avaliation view', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Avaliation />).toJSON() 
    expect(tree).toMatchSnapshot()
    })
    describe('test button PUBLICA and ANONIMA', () => {
        it('children´s of the buttons must be the same on click them', () => {
            render(<Avaliation />);
            const publica = { 
                button: screen.getByText('PÚBLICA'),
                old_children: screen.getByText('PÚBLICA').children
            }
            const anonima = { 
                button: screen.getByText('ANÔNIMA'),
                old_children: screen.getByText('ANÔNIMA').children
            }

            userEvent.click(publica.button);
            expect(publica.old_children).toBe(publica.button.children);

            userEvent.click(anonima.button);
            expect(anonima.old_children).toBe(anonima.button.children);
        })
    })
    // describe('test of submit form', () => {
    //     it('data must be post in api', done => {
    //         const data = {
    //             reg_student: "123456789"
    //         }
    //         const professor = {
    //             id_professor: '',
    //             disciplines: [{
    //                 discipline_code: 1,
    //                 name: "teste1"
    //             },{
    //                 discipline_code: 2,
    //                 name: "teste2"
    //             }]
    //         }
    //         const callback = jest.fn();
    //         localStorage.setItem('student', data);
    //         render(<Avaliation close={callback} professor={professor}/>);
    //         const input = screen.getByTestId('form-aval-1');

    //         const select = screen.getByTestId('select-aval-1');
    //         userEvent.click(select.children[1]);

    //         const star = screen.getByTestId('star-aval-1');
    //         userEvent.click(star.children[4].children[0])
            
    //         const star1 = screen.getByTestId('star-aval-2');
    //         userEvent.click(star1.children[4].children[0])

    //         const star2 = screen.getByTestId('star-aval-3');
    //         userEvent.click(star2.children[4].children[0])

    //         const star3 = screen.getByTestId('star-aval-4');
    //         userEvent.click(star3.children[4].children[0])

    //         const anonima = screen.getByText('ANÔNIMA');
    //         userEvent.click(anonima);

    //         const descricao = screen.getByTestId('form-aval-2');
    //         userEvent.type(descricao, "TESTE DE POST");

    //         fireEvent.submit(input);
      
    //         expect(callback).toHaveBeenCalled();
    //     })
    // })
})