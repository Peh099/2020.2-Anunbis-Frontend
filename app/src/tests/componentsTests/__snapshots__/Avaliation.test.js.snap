// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Avaliation view matches the snapshot 1`] = `
<div
  className="sc-crzoAE feGJIF"
>
  <div
    className="sc-dIsUp hJMXzL"
  >
    <div
      className="sc-bqGGPW dAWHJi"
    >
      Avaliação
    </div>
    <button
      className="sc-jSFjdj iYEbuK"
      radius="0px"
    >
      X
    </button>
  </div>
  <main
    className="sc-ksluID dqykrH"
  >
    <div
      className="sc-bdnxRM kjrnMz"
    >
      <form
        className="sc-kEqXSa cgEWQw"
        data-testid="form-aval-1"
        onSubmit={[Function]}
      >
        <p
          className="sc-dlnjwi djLcli"
        />
        <div
          className="sc-pNWdM eUdqmP"
        >
          <select
            className="sc-eCApnc jaaHNR"
            id="diciplines"
            name="id_course"
          >
            <option
              disabled={true}
              selected={true}
              value=""
            >
              Selecione um Curso
            </option>
          </select>
          <div
            className="sc-jrsJWt fvYlXW error"
          />
        </div>
        <p>
          Notas:
        </p>
        <div
          className="sc-gtsrHT kavaLt"
        >
          <label>
            Didática:
          </label>
          <div
            className="sc-pNWdM eUdqmP"
          >
            <div
              className="sc-gKAaRy cgyNEA"
            >
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-1"
                  name="didactic"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-2"
                  name="didactic"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-3"
                  name="didactic"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-4"
                  name="didactic"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-5"
                  name="didactic"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
            </div>
            <div
              className="sc-jrsJWt fvYlXW error"
            />
          </div>
          <label>
            Metodologia:
          </label>
          <div
            className="sc-pNWdM eUdqmP"
          >
            <div
              className="sc-gKAaRy cgyNEA"
            >
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-1"
                  name="metod"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-2"
                  name="metod"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-3"
                  name="metod"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-4"
                  name="metod"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-5"
                  name="metod"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
            </div>
            <div
              className="sc-jrsJWt fvYlXW error"
            />
          </div>
          <label>
            Coerência das Avaliações:
          </label>
          <div
            className="sc-pNWdM eUdqmP"
          >
            <div
              className="sc-gKAaRy cgyNEA"
            >
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-1"
                  name="avaliations"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-2"
                  name="avaliations"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-3"
                  name="avaliations"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-4"
                  name="avaliations"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-5"
                  name="avaliations"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
            </div>
            <div
              className="sc-jrsJWt fvYlXW error"
            />
          </div>
          <label>
            Disponibilidade:
          </label>
          <div
            className="sc-pNWdM eUdqmP"
          >
            <div
              className="sc-gKAaRy cgyNEA"
            >
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-1"
                  name="disponibility"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-2"
                  name="disponibility"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-3"
                  name="disponibility"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-4"
                  name="disponibility"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
              <label
                className="sc-fujyAs exNMkG"
              >
                <input
                  checked={false}
                  data-testid="input-stars-5"
                  name="disponibility"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="sc-iCoGMd iNWuFv"
                />
              </label>
            </div>
            <div
              className="sc-jrsJWt fvYlXW error"
            />
          </div>
        </div>
        <p>
          Postagem:
        </p>
        <div
          className="sc-bdnxRM fGJwKM"
          direction="row"
        >
          <button
            className="sc-jSFjdj jYSBhf"
            onClick={[Function]}
            type="button"
          >
            PÚBLICA
          </button>
          <button
            className="sc-jSFjdj fTYZxU"
            onClick={[Function]}
            type="button"
          >
            ANÔNIMA
          </button>
        </div>
        <p>
          Descrição/Comentários:
        </p>
        <div
          className="sc-pNWdM eUdqmP"
        >
          <textarea
            className="sc-hKFxyN cqWcJi"
            data-testid="form-aval-2"
            name="comments"
          />
          <div
            className="sc-jrsJWt fvYlXW error"
          />
        </div>
        <div
          className="sc-iqAclL fwgQfe"
        >
          <button
            className="sc-jSFjdj iooaEn"
            type="submit"
          >
            POSTAR
          </button>
        </div>
      </form>
    </div>
  </main>
</div>
`;
